### PR TITLE
[native_assets_builder] Fix old build hooks

### PR DESCRIPTION
**Pull request by dcharkes**
When a newer Dart/Flutter SDK invokes an old build hook, the old build hook will write `build_output.yaml` instead of `build_output.json`. To deal with version skew where the SDK is new and the build hook is old, fall back to the yaml file if the json file doesn't exist.

Adds a new integration test which contains a build hook that uses an old version of `package:native_assets_cli`.

Issue:

* https://github.com/dart-lang/sdk/issues/55712

### CODE REVIEW

File: pkgs/native_assets_builder/test_data/native_add_v1_0_0/src/native_add.c [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+#include "native_add.h"
+
+int32_t add(int32_t a, int32_t b) {
```
Comment (HosseinYousefi) [inline pkgs/native_assets_builder/test_data/native_add_v1_0_0/src/native_add.c:7]: Most written piece of C code for testing Dart FFI!

**Review by HosseinYousefi — APPROVED** [review]
LGTM!

File: pkgs/native_assets_builder/test_data/native_add_v1_0_0/src/native_add.c (hunk 1 above)
Comment (dcharkes) [reply pkgs/native_assets_builder/test_data/native_add_v1_0_0/src/native_add.c:7]: Correction, most copied piece of code 🤣

**Comment by dcharkes** [comment]
I don't know why the windows coverage hangs. Merging.

**Pull request merged**